lib/matplotlib/axes/_base.py

```diff
@@ -2512,6 +2517,50 @@ def handle_single_axis(scale, autoscaleon, shared_axes, interval,
     def _get_axis_list(self):
         return (self.xaxis, self.yaxis)
 
+    def _update_title_position(self, renderer):
+        """
+        Update the title position based on the bounding box enclosing
+        all the ticklabels and x-axis spine and xlabel...
+        """
+        _log.debug('update_title_pos')
+
+        if self._autotitlepos is not None and not self._autotitlepos:
+            _log.debug('title position was updated manually, not adjusting')
+            return
+
+        titles = (self.title, self._left_title, self._right_title)
+
+        if self._autotitlepos is None:
+            for title in titles:
+                x, y = title.get_position()
+                if not np.isclose(y, 1.0):
+                    self._autotitlepos = False
+                    _log.debug('not adjusting title pos because title was'
+                             ' already placed manually: %f', y)
+                    return
+            self._autotitlepos = True
+
+        for title in titles:
+            x, y0 = title.get_position()
+            y = 1.0
+            # need to check all our twins too...
+            axs = self._twinned_axes.get_siblings(self)
+
+            for ax in axs:
+                try:
+                    if (ax.xaxis.get_label_position() == 'top'
+                            or ax.xaxis.get_ticks_position() == 'top'):
+                        bb = ax.xaxis.get_tightbbox(renderer)
+                        top = bb.y1
```

Comment:
Should it not be `top = bb.ymax` rather than `top = bb.y1`? That is a very genuine question, because I am not that used to play with boundary boxes and the docstring of `y1` reads
```
:attr:`y1` is the second of the pair of *y* coordinates that
define the bounding box. :attr:`y1` is not guaranteed to be greater
than :attr:`y0`.  If you require that, use :attr:`ymax`.
```

Reply:
Oh, cool, I didn't know that.

Reply:
Do we prefer to stay with `top = bb.y1`, while it actually means `top = bb.ymax` (if I correctly understood the process)? I guess that in 99.99% of the cases here, it will be equivalent, but the semantic of the latter might be clearer, right? (I have no strong feelings about any of the two options)